iroh-net/src/net/interfaces/bsd.rs

```diff
@@ -374,9 +374,9 @@ impl WireFormat {
                     return Err(RouteError::InvalidMessage);
                 }
 
-                #[cfg(target_arch = "netbsd")]
+                #[cfg(target_os = "netbsd")]
                 let index = u16_from_ne_range(data, 16..18)?;
-                #[cfg(not(target_arch = "netbsd"))]
+                #[cfg(not(target_os = "netbsd"))]
                 let index = u16_from_ne_range(data, 12..14)?;
```

Comment:
Looks like this was [introduced](https://github.com/n0-computer/iroh/pull/1472/files#diff-75f83c1e913d60def929bd1986bb190036c81a21cbf37b384c11b5251347503fR290-R293) in https://github.com/n0-computer/iroh/pull/1472 by @dignifiedquire.

Given we've been running with this for a while now, do you know if "fixing" this cfg will actually make things better, not worse? @dignifiedquire

Reply:
I suspect we just never hit this branch in practice yet, but the cfg is the only part that's wrong asfatict